File: Challenge/Demo2/Backend/src/main/java/com/task/domain/Task.java
package com.task.domain;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "tasks")
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    
    @Column(nullable = false)
    private String title;
    
    private String description;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskStatus status = TaskStatus.TODO;
    
    private LocalDateTime deadline;
    
    @Column(nullable = false)
    private String assignee;
    
    @Column(nullable = false)
    private String createdBy;
    
    private String project;
    
    @ElementCollection
    @CollectionTable(name = "task_labels", joinColumns = @JoinColumn(name = "task_id"))
    @Column(name = "label")
    private List<String> labels;
    
    @Enumerated(EnumType.STRING)
    private Priority priority;
    
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "task_id")
    private List<Reminder> reminders;
    
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "task_id")
    private List<Comment> comments;
    
    private LocalDateTime deletedAt;
    
    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: Challenge/Demo2/Frontend/src/types/Task.ts
export interface Reminder {
  id: string;
  time: string;
  message: string;
}

export interface Comment {
  id: string;
  content: string;
  createdAt: string;
  createdBy: string;
}

export interface Task {
  id: string;
  title: string;
  description?: string;
  status: 'todo' | 'inprogress' | 'done';
  deadline?: string;
  assignee: string;
  createdBy: string;
  project?: string;
  labels?: string[];
  priority?: 'low' | 'medium' | 'high';
  reminders?: Reminder[];
  comments?: Comment[];
  deletedAt?: string;
}
